tests/machine_control/test_machine_controller.py

```diff
@@ -539,6 +539,46 @@ def test_app_stop_and_count(self, controller):
                 (_, app_id, _) = entry
                 assert app_id == 0
 
+    @pytest.mark.order_after("live_test_load_application")
+    def test_sdram_alloc_and_free(self, controller):
+        """Test subsequent allocation, freeing and reallocation of a large
+        block of memory.
+
+        This is an implicit test that FREE actually works.
+        """
+        with controller(x=0, y=0):
+            # Allocate 90MiB
+            ptr = controller.sdram_alloc(90*(1 << 20))
```

Comment:
Does this not require an assertion to make sure the allocation actually occurred?

(I would probably also include a second alloc which you assert for failure, as a sanity check).